translib/transformer/subscribe_req_xlate.go

```diff
@@ -0,0 +1,2135 @@
+	}
+
+	subReqXlator := subscribeReqXlator{subReq: &subReq}
+	subReqXlator.subReq.subReqXlateInfo = new(XfmrSubscribeReqXlateInfo)
+	subReqXlator.pathXlator = &subscribePathXlator{subReq: &subReq}
+	return &subReqXlator, nil
+}
+
+func (reqXlator *subscribeReqXlator) processSubscribePath() error {
+	if log.V(dbLgLvl) {
+		log.Infof("%v processSubscribePath: path: %v", reqXlator.subReq.reqLogId, reqXlator.subReq.reqUri)
+	}
+	pathElems := reqXlator.subReq.gPath.Elem
+	pathIdx := len(pathElems) - 1
+	ygXpathInfoTmp := reqXlator.subReq.subReqXlateInfo.ygXpathTrgtInfo
+	ygEntryTmp, ygErr := getYgEntry(reqXlator.subReq.reqLogId, ygXpathInfoTmp, reqXlator.subReq.ygPath)
+	if ygErr != nil {
+		return ygErr
+	}
+	ygPathTmp := reqXlator.subReq.ygPath
+	isUriMdfd := false
+	for {
+		if ygEntryTmp.Parent != nil && ygXpathInfoTmp.nameWithMod != nil {
+			if log.V(dbLgLvl) {
+				log.Infof("%v parent node prefix: %v and curr. node prefix: %v", reqXlator.subReq.reqLogId, ygEntryTmp.Parent.Prefix.Name,
+					ygEntryTmp.Prefix.Name)
+			}
+			if strings.HasSuffix(*ygXpathInfoTmp.nameWithMod, ":"+pathElems[pathIdx].Name) &&
+				ygEntryTmp.Parent.Prefix.Name != ygEntryTmp.Prefix.Name {
+				if log.V(dbLgLvl) {
+					log.Infof("%v module prefix is missing in the path: adding the same: mod prefix: %v, "+
+						"input path name: %v", reqXlator.subReq.reqLogId, *ygXpathInfoTmp.nameWithMod, pathElems[pathIdx].Name)
+				}
+				pathElems[pathIdx].Name = *ygXpathInfoTmp.nameWithMod
+				isUriMdfd = true
+			}
+		}
+
+		if ygEntryTmp.IsList() && len(pathElems[pathIdx].Key) == 0 {
+			isUriMdfd = true
+			for _, listKey := range strings.Fields(ygEntryTmp.Key) {
+				pathElems[pathIdx].Key[listKey] = "*"
+			}
+			if log.V(dbLgLvl) {
+				log.Infof("%v processSubscribePath: list node doesn not have keys in the input path;"+
+					" added wildcard to the list node path: %v", reqXlator.subReq.reqLogId, reqXlator.subReq.ygPath)
+			}
+		} else if reqXlator.subReq.isTrgtPathWldcrd {
+			for _, kv := range pathElems[pathIdx].Key {
+				if kv == "*" {
+					continue
+				}
+				reqXlator.subReq.isTrgtPathWldcrd = false
+				break
+			}
+		}
+
+		tIdx := strings.LastIndex(ygPathTmp, "/")
+		if tIdx > 0 {
+			ygPathTmp = ygPathTmp[0:tIdx]
+			pathIdx--
+			if log.V(dbLgLvl) {
+				log.Infof("%v processSubscribePath: xPathTmp: %v", reqXlator.subReq.reqLogId, ygPathTmp)
+			}
+			var ok bool
+			ygXpathInfoTmp, ok = xYangSpecMap[ygPathTmp]
+			if ok && ygXpathInfoTmp != nil {
+				var ygErr error
+				if ygEntryTmp, ygErr = getYgEntry(reqXlator.subReq.reqLogId, ygXpathInfoTmp, ygPathTmp); ygErr != nil {
+					return ygErr
+				}
+			}
+			if !ok || ygXpathInfoTmp == nil || ygEntryTmp == nil {
+				log.Warning(reqXlator.subReq.reqLogId, "processSubscribePath: xYangSpecMap does not have the yangXpathInfo for the path:", ygPathTmp)
+				return tlerr.InternalError{Format: "xYangSpecMap does not have the yangXpathInfo", Path: ygPathTmp}
+			}
+		} else {
+			break
```

Comment:
Consider breaking in the negative condition:

if tIdx <= 0 {
    break
}
ygPathTmp = ygPathTmp[0:tIdx]
...